### PR TITLE
Fix broken tests on CI/CD

### LINKED ISSUE #492: FIX: Broken Tests to Unblock CI/CD
### Failing tests:

Encountered 1 failing test in test/dollar/Staking.t.sol:DepositStateTest
- [ ] FAIL. Reason: Setup failed: EvmError: Revert setUp() (gas: 0)

Encountered 1 failing test in test/dollar/Staking.t.sol:ZeroStateTest
- [ ] FAIL. Reason: Setup failed: EvmError: Revert setUp() (gas: 0)

Encountered 1 failing test in test/dollar/StakingShare.t.sol:DepositStateTest
- [ ] FAIL. Reason: Setup failed: EvmError: Revert setUp() (gas: 0)

Encountered 1 failing test in test/dollar/UbiquityChef.t.sol:DepositStateTest
- [ ] FAIL. Reason: Setup failed: EvmError: Revert setUp() (gas: 0)

Encountered 1 failing test in test/dollar/UbiquityChef.t.sol:ZeroStateTest
- [ ] FAIL. Reason: Setup failed: EvmError: Revert setUp() (gas: 0)

Encountered 1 failing test in test/manager/facets/ManagerFacet.t.sol:TestManagerFacet
- [ ] FAIL. Reason: EvmError: Revert testShouldDeployStableSwapPool() (gas: 2232362)

Encountered 5 failing tests in test/ubiquistick/UbiquiStick.t.sol:UbiquiStickTest
- [ ] FAIL. Reason: Call did not revert as expected testApproveForBannedOperator() (gas: 170582)
- [ ] FAIL. Reason: Call did not revert as expected testSafeTransferFromForBannedOperator() (gas: 170580)
- [ ] FAIL. Reason: Call did not revert as expected testSafeTransferFromWithDataForBannedOperator() (gas: 170536)
- [ ] FAIL. Reason: Call did not revert as expected testSetApprovalsAllForBannedOperator() (gas: 170094)
- [ ] FAIL. Reason: Call did not revert as expected testTransferFromForBannedOperator() (gas: 170536)

```
Encountered a total of 11 failing tests, 156 tests succeeded
command closing on process
/home/runner/work/ubiquity-dollar/ubiquity-dollar/packages/contracts/scripts/runner/runner.ts:28
      throw new Error(`Failing tests ${code}`);
            ^
Error: Failing tests 1
    at ChildProcess.<anonymous> (/home/runner/work/ubiquity-dollar/ubiquity-dollar/packages/contracts/scripts/runner/runner.ts:28:13)
    at ChildProcess.emit (node:events:513:28)
    at ChildProcess.emit (node:domain:489:12)
    at maybeClose (node:internal/child_process:1091:16)
    at ChildProcess._handle.onexit (node:internal/child_process:302:5)
Node.js v18.13.0
```

### Related Logs

```
ERROR forge::runner setUp failed reason="EvmError: Revert" contract=0x7fa9385be102ac3eac297483dd6233d62b3e1496
Running 1 test for test/dollar/Staking.t.sol:DepositStateTest
FAIL. Reason: Setup failed: EvmError: Revert setUp() (gas: 0)
Test result: FAILED. 0 passed; 1 failed; finished in 8.24ms
ERROR forge::runner setUp failed reason="EvmError: Revert" contract=0x7fa9385be102ac3eac297483dd6233d62b3e1496
Running 1 test for test/dollar/Staking.t.sol:ZeroStateTest
FAIL. Reason: Setup failed: EvmError: Revert setUp() (gas: 0)
Test result: FAILED. 0 passed; 1 failed; finished in 8.17ms
ERROR forge::runner setUp failed reason="EvmError: Revert" contract=0x7fa9385be102ac3eac297483dd6233d62b3e1496
Running 1 test for test/dollar/StakingShare.t.sol:DepositStateTest
FAIL. Reason: Setup failed: EvmError: Revert setUp() (gas: 0)
Test result: FAILED. 0 passed; 1 failed; finished in 8.44ms
Running 11 tests for test/ubiquistick/UbiquiStick.t.sol:UbiquiStickTest
FAIL. Reason: Call did not revert as expected testApproveForBannedOperator() (gas: 170582)
FAIL. Reason: Call did not revert as expected testSafeTransferFromForBannedOperator() (gas: 170580)
FAIL. Reason: Call did not revert as expected testSafeTransferFromWithDataForBannedOperator() (gas: 170536)
FAIL. Reason: Call did not revert as expected testSetApprovalsAllForBannedOperator() (gas: 170094)
FAIL. Reason: Call did not revert as expected testTransferFromForBannedOperator() (gas: 170536)
Test result: FAILED. 6 passed; 5 failed; finished in 2.71ms
ERROR forge::runner setUp failed reason="EvmError: Revert" contract=0x7fa9385be102ac3eac297483dd6233d62b3e1496
Running 1 test for test/dollar/UbiquityChef.t.sol:DepositStateTest
FAIL. Reason: Setup failed: EvmError: Revert setUp() (gas: 0)
Test result: FAILED. 0 passed; 1 failed; finished in 8.26ms
ERROR forge::runner setUp failed reason="EvmError: Revert" contract=0x7fa9385be102ac3eac297483dd6233d62b3e1496
Running 1 test for test/dollar/UbiquityChef.t.sol:ZeroStateTest
FAIL. Reason: Setup failed: EvmError: Revert setUp() (gas: 0)
Test result: FAILED. 0 passed; 1 failed; finished in 7.97ms
Running 17 tests for test/dollar/ERC4626.t.sol:ERC4626Test
Running 20 tests for test/manager/facets/ManagerFacet.t.sol:TestManagerFacet
FAIL. Reason: EvmError: Revert testShouldDeployStableSwapPool() (gas: 2232362)
Test result: FAILED. 19 passed; 1 failed; finished in 7.60ms
```

### PR DESCRIPTION
**Pull request by Steveantor**
- Update testrunner
- Update test files

Resolves #492

### CODE REVIEW

File: packages/contracts/test/dollar/UbiquityChef.t.sol [hunk 1]
```diff
@@ -6,11 +6,15 @@ import "../helpers/LiveTestHelper.sol";
...
+        address indexed user,
+        uint256 amount,
+        uint256 indexed stakingShareId
     );
 
     event Withdraw(
-        address indexed user, uint256 amount, uint256 indexed stakingShareId
+        address indexed user,
+        uint256 amount,
+        uint256 indexed stakingShareId
```
Comment (0x4007) [inline packages/contracts/test/dollar/UbiquityChef.t.sol:17]: What's up with the formatting?

**Review by 0x4007 — CHANGES_REQUESTED** [review]
I thought we had the formatters (prettier, eslint) as part of our CI? If not we should create a new bounty for it. 

We shouldn't be accepting formatting changes unless they match our settings. I might let this one slide but @Steveantor please confirm that you used the correct Solidity formatting settings here.

File: packages/contracts/test/dollar/UbiquityChef.t.sol (hunk 1 above)
Comment (Steveantor) [reply packages/contracts/test/dollar/UbiquityChef.t.sol:17]: I have my Solidity formatter disabled, it's likely the output of 
Lint-Staged @pavlovcik

**Comment by Steveantor** [comment]
> I thought we had the formatters (prettier, eslint) as part of our CI? If not we should create a new bounty for it.
> 
> We shouldn't be accepting formatting changes unless they match our settings. I might let this one slide but @Steveantor please confirm that you used the correct Solidity formatting settings here.

Lint-Staged confirmed

> image
> Before commit

> image
> After commit

File: packages/contracts/test/dollar/UbiquityChef.t.sol (hunk 1 above)
Comment (0x4007) [reply packages/contracts/test/dollar/UbiquityChef.t.sol:17]: This implies that we don't have our entire codebase formatted. We should run a quick formatting across the entire codebase then!

Comment (Steveantor) [reply packages/contracts/test/dollar/UbiquityChef.t.sol:17]: > This implies that we don't have our entire codebase formatted. We should run a quick formatting across the entire codebase then!

that's right, our contracts aren't following the Prettier standard.

**Review by 0x4007 — APPROVED** [review]

**Comment by 0x4007** [comment]
Also I don't understand why install & build CI is failing now but it doesn't seem to be from this PR so I'll merge.

**Comment by rndquu** [comment]
@Steveantor 

why should we divide tests for "local" and "remote"? Can't we use `forge test --rpc-url OPTIMAL_URL` for all tests?

I've just tried `forge test --rpc-url https://cloudflare-eth.com` and all tests passed as expected. Right now tests are failing in the `development` branch because `--rpc-url` option is missed. Can we simply add that option for tests without dividing tests for "local" and "remote"?

**Comment by Steveantor** [comment]
> @Steveantor
> 
> why should we divide tests for "local" and "remote"? Can't we use `forge test --rpc-url OPTIMAL_URL` for all tests?
> 
> I've just tried `forge test --rpc-url https://cloudflare-eth.com` and all tests passed as expected. Right now tests are failing in the `development` branch because `--rpc-url` option is missed. Can we simply add that option for tests without dividing tests for "local" and "remote"?

Most tests run fine locally but some tests require dynamic/externally fetched block data,
which the local testnet couldn't simply handle.

Running all tests on RPC quickly hits up the rate limit, and GA uses shared IPs on top of that.
That's why we should only run the required/minimal amount of tests remotely,
Local tests are also pretty faster.

**Pull request merged**